### DISCUSSION TITLE
Add stylus based notes functionality

### DISCUSSION THREAD

**Discussion by sanjarcode** [zim-desktop-wiki/zim-desktop-wiki#2052] [Ideas]
**Use case**
Add editable handwritten notes using a stylus. **Not** a handwriting to text converter. 

**Plugin Idea**
* I sometimes feel the need to embed short handwritten notes in zim, like an image.
* [Write](http://www.styluslabs.com/) is a free, lightweight *stylus* based note-taking app which works on all major OSs. 
* All notes in Write are saved as an SVG and HTML file. The SVG can be inserted as an image in zim.
* On clicking(or some other activation) the image, a Write window can open the corresponding HTML(name same as that of the SVG), in edit mode. This will be quick I suppose https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/308#issue-293970056
* Fortunately, it's not necessary to *install* Write to *view* the notes, as they are SVGs. This makes viewing shared notes very easy.

**Alternatives considered**
1. A simple solution would be to write the stuff on paper/tablet and insert the screenshot/photo. Such <q>notes</q> are not editable. 
2. Xournal is an app for this purpose but it's very big compared to Write, and makes the whole setup heavy. An old issue: https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/559#issue-380791041

**Plan**
1. Option to attach notes.
2. Option to edit notes.

![image](https://user-images.githubusercontent.com/48156230/84904041-7cb0cd00-b0cc-11ea-833f-932aff5cf2d5.png)

Any suggestions? I'm making a zim plugin for the first time.

**Comment by introt** [comment]
I've been thinking about writing a plugin to wrap Xournal++ in the style of Insert Equation - storing the notebook as an attachment and previewing it as a png or svg. Xournal++ can already export transparent pngs and I've been using ImageMagick's trim to remove the empty edges. Haven't looked at Write. Keep us posted, looking forward to your implementation!

**Comment by sanjarcode** [comment]
I've already contacted Write devs. Attaching their response.
[Gmail - Is Write free and_or opensource_ Wanted to integrate with zim-wiki.pdf](https://github.com/zim-desktop-wiki/zim-desktop-wiki/files/6568994/Gmail.-.Is.Write.free.and_or.opensource_.Wanted.to.integrate.with.zim-wiki.pdf)

My stylus broke soon after this conversation last year.
I'll buy one and get to it ASAP, 🚀️.
